app_spec/test/files/offline-validation.js

```diff
@@ -0,0 +1,107 @@
+        let chain_headers = []
+
+        await s.consistency()
+
+        for (let i=0; i< chain_header_hashes.length; i++) {
+            // can use get_post because it just returns a raw entry given a hash
+            let header_hash = chain_header_hashes[i]
+            t.comment(header_hash)
+
+            // check alice can retrieve their own header entries
+            let header_alice = await alice.call('app', "blog", "get_post", { post_address: header_hash })
+            t.ok(header_alice.Ok)
+
+            // check bob can retrieve alices header entries
+            let header_bob = await bob.call('app', "blog", "get_post", { post_address: header_hash })
+            t.ok(header_bob.Ok)
+
+            t.deepEqual(header_alice.Ok, header_bob.Ok)
+
+            chain_headers.push(header_bob.Ok)
+        }
+        t.comment(JSON.stringify(chain_headers))
+      })
+
+      scenario('Can perform validation of an entry while the author is offline', async (s, t) => {
+        
+        const { alice, bob, carol } = await s.players({alice: one, bob: one, carol: one})
+        // alice and bob start online
+        await alice.spawn()
+        await s.consistency()
+        await bob.spawn()
+        await s.consistency()
+
+        // alice publishes the original entry. !This is an entry that requires full chain validation!
+        const initialContent = "Holo world y'all"
+        const params = { content: initialContent, in_reply_to: null }
+        const create_result = await alice.call('app', "blog", "create_post", params)
+        t.comment(JSON.stringify(create_result))
+        t.equal(create_result.Ok.length, 46)
+        await s.consistency()
+
+        t.comment('waiting for consistency between Alice and Bob')
+        // bob will receive the entry and hold it
+        await s.consistency()
+        t.comment('consistency has been reached')
+
+        // check bob got the content Ok
+        const bob_result = await bob.call('app', "blog", "get_post", { post_address: create_result.Ok })
+        t.ok(bob_result.Ok)
+        t.equal(JSON.parse(bob_result.Ok.App[1]).content, initialContent)
+        
+        // alice then goes offline
+        t.comment('waiting for alice to go offline')
+        await alice.kill()
+        t.comment('alice has gone offline')
+
+        // carol then comes online, will receive the entry via gossip from bob and need to validate
+        // Since alice is offline the validation package cannot come direct and must
+        // be regenerated from the published headers (which bob should hold)
+        t.comment('waiting for Carol to come online')
+        await carol.spawn()
+        t.comment('Carol is online')
+
+        t.comment('Waiting for Carol to get all data via gossip')
+        await s.consistency()
+        t.comment('consistency has been reached')
+
+        await delay(50000) // keep this until consistency works with dynamic starting agents
```

Comment:
The reason it has to be quite long is because it waits a full 10 seconds for the author to reply with their chain before falling back to rebuilding it. This delay just needs to be large relative to that. From my experiments locally we can't make it much shorter without introducing flakyness

Reply:
As soon as it becomes high enough priority, we can update hachiko's NetworkModel to account for dynamically changing network membership. The change can probably be done in half a day. Maybe it's time?

Reply:
does https://github.com/holochain/holochain-rust/pull/1957 allow us to remove this giant delay?